src/Parallel/ResourceInfo.hpp

```diff
@@ -242,10 +247,30 @@ bool operator!=(const SingletonPack<tmpl::list<Components...>>& lhs,
 }
 
 namespace detail {
+// This whole gymnastics with type aliases is necessary because ResourceInfo is
+// inside the GlobalCache. There are a lot of places that use the GlobalCache
+// that haven't defined the initialization_tags type alias (like the testing
+// framework). And even though we don't call the entry method on the GlobalCache
+// related to ResourceInfo, all the type aliases inside ResourceInfo are still
+// constructed regardless. So rather than changing every mock component to have
+// an empty type alias, we just treat no type alias as not having the tag we are
+// looking for.
+CREATE_HAS_TYPE_ALIAS(initialization_tags)
+CREATE_HAS_TYPE_ALIAS_V(initialization_tags)
+
+template <typename Component, typename Tag, bool HasInitializationTags>
```

Comment:
you can just default the last argument to `bool HasInitializationTags = has_initialization_tags_v<Component>` and then not specify it everywhere else